Assets/MixedRealityToolkit.Tests/EditModeTests/Core/TestFixture_02_MixedRealityToolkitFilesTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.MixedReality.Toolkit.Tests.Core
+{
+    // Tests for the MixedRealityToolkitFiles utility class
+    public class TestFixture_02_MixedRealityToolkitFilesTests
+    {
+        [Test]
+        public void Test_01_FindMatchingModule()
+        {
+            TestInvalidPath("");
+
+            // Test invalid base name
+            TestInvalidPath("aaa");
+            TestInvalidPath(".SDK");
+            TestInvalidPath("aaa.SDK");
+
+            // Test missing chars
+            TestInvalidPath("MixedRealityToolki.SDK");
+            TestInvalidPath("MixedRealityToolkit.SD");
+            TestInvalidPath("ixedRealityToolkit.SDK");
+            TestInvalidPath("MixedRealityToolkit.DK");
+
+            // Test missing dots
+            TestInvalidPath("SDK");
+            TestInvalidPath("MixedRealityToolkitSDK");
+
+            // Test valid paths
+            TestValidPath("MixedRealityToolkit", MixedRealityToolkitModuleType.Core);
+            TestValidPath("MixedRealityToolkit.SDK", MixedRealityToolkitModuleType.SDK);
+
+            var modules = Enum.GetValues(typeof(MixedRealityToolkitModuleType));
```

Comment:
Nit: Add a comment here saying that the intent is to ensure that every single enum value is properly handled by the Dictionary in MRTKFiles - in other words, if this fails, the likely cause is that you haven't updated that dictionary. This provides direct action when someone adds an enum later and tells the why the thing below is broken and how to fix it.